================================================== cannot_call_non_func
exit_code: 103
--------------------------------------------------
"test"();
--------------------------------------------------
--------------------------------------------------
runtime_errors/cannot_call_non_func.sd:1:1: can't call 'string' as a function
================================================== undefined
exit_code: 103
--------------------------------------------------
not_defined;
--------------------------------------------------
--------------------------------------------------
runtime_errors/undefined.sd:1:1: 'not_defined' is not defined
================================================== undefined_assignment
exit_code: 103
--------------------------------------------------
not_defined = 3;
--------------------------------------------------
--------------------------------------------------
runtime_errors/undefined_assignment.sd:1:1: 'not_defined' is not defined
================================================== invalid_bind_target
exit_code: 103
--------------------------------------------------
1 := 1;
--------------------------------------------------
--------------------------------------------------
runtime_errors/invalid_bind_target.sd:1:1: cannot bind to an integer literal
================================================== already_in_scope1
exit_code: 103
--------------------------------------------------
x := 1;
x := 1;
--------------------------------------------------
--------------------------------------------------
runtime_errors/already_in_scope1.sd:2:1: 'x' is already defined in the current scope at [1:1]
================================================== already_in_scope2
exit_code: 103
--------------------------------------------------
x := 1;
fn x(){}
--------------------------------------------------
--------------------------------------------------
runtime_errors/already_in_scope2.sd:2:4: 'x' is already defined in the current scope at [1:1]
================================================== already_in_scope3
exit_code: 103
--------------------------------------------------
fn x(){}
x := 1;
--------------------------------------------------
--------------------------------------------------
runtime_errors/already_in_scope3.sd:2:1: 'x' is already defined in the current scope at [1:4]
================================================== arg_num_mismatch
exit_code: 103
--------------------------------------------------
fn f(a, b){}
f(1);
--------------------------------------------------
--------------------------------------------------
runtime_errors/arg_num_mismatch.sd:2:1: expected 2 arguments, got 1
================================================== incorrect_type
exit_code: 103
--------------------------------------------------
{1: ""};
--------------------------------------------------
--------------------------------------------------
runtime_errors/incorrect_type.sd:1:2: property name must be 'string', got 'int'
================================================== dup_arg_names
exit_code: 103
--------------------------------------------------
fn f(a, a){}
f(1, 2);
--------------------------------------------------
--------------------------------------------------
runtime_errors/dup_arg_names.sd:1:9: in 'f': 'a' is already defined in the current scope at [1:6]
Stacktrace:
  runtime_errors/dup_arg_names.sd:2:1: in '<root>'
================================================== add_bad_types1
exit_code: 103
--------------------------------------------------
true + {};
--------------------------------------------------
--------------------------------------------------
runtime_errors/add_bad_types1.sd:1:6: can't apply '+' to 'bool' and 'object'
================================================== add_bad_types2
exit_code: 103
--------------------------------------------------
1 + 2 * [];
--------------------------------------------------
--------------------------------------------------
runtime_errors/add_bad_types2.sd:1:7: can't apply '*' to 'int' and 'list'
================================================== bad_type_equality
exit_code: 103
--------------------------------------------------
[] == {};
--------------------------------------------------
--------------------------------------------------
runtime_errors/bad_type_equality.sd:1:4: can't apply '==' to 'list' and 'object'
================================================== op_assign_lhs_not_var
exit_code: 103
--------------------------------------------------
1 += 1;
--------------------------------------------------
--------------------------------------------------
runtime_errors/op_assign_lhs_not_var.sd:1:1: the LHS of an operation-assignment must be a variable
================================================== op_assign_bad_types
exit_code: 103
--------------------------------------------------
x := 1;
x += "";
--------------------------------------------------
--------------------------------------------------
runtime_errors/op_assign_bad_types.sd:2:3: can't apply '+' to 'int' and 'string'
================================================== return_outside_function
exit_code: 103
--------------------------------------------------
return null;
--------------------------------------------------
--------------------------------------------------
runtime_errors/return_outside_function.sd:1:1: 'return' can't be used outside of a function
================================================== value_not_indexable
exit_code: 103
--------------------------------------------------
1[2];
--------------------------------------------------
--------------------------------------------------
runtime_errors/value_not_indexable.sd:1:1: only 'list's, 'object's or 'string's can be indexed
================================================== out_of_string_bounds
exit_code: 103
--------------------------------------------------
"abc"[5];
--------------------------------------------------
--------------------------------------------------
runtime_errors/out_of_string_bounds.sd:1:1: index '5' is outside the string bounds
================================================== out_of_list_bounds
exit_code: 103
--------------------------------------------------
xs := [1, 2, 3];
xs[2];
xs[3];
--------------------------------------------------
--------------------------------------------------
runtime_errors/out_of_list_bounds.sd:3:1: index '3' is outside the list bounds
================================================== out_of_list_bounds_assign
exit_code: 103
--------------------------------------------------
[1, 2, 3][3] = 2;
--------------------------------------------------
--------------------------------------------------
runtime_errors/out_of_list_bounds_assign.sd:1:1: index '3' is outside the list bounds
================================================== prop_not_found1
exit_code: 103
--------------------------------------------------
{}["a"];
--------------------------------------------------
--------------------------------------------------
runtime_errors/prop_not_found1.sd:1:1: object doesn't contain property 'a'
================================================== prop_not_found2
exit_code: 103
--------------------------------------------------
{}.a;
--------------------------------------------------
--------------------------------------------------
runtime_errors/prop_not_found2.sd:1:1: object doesn't contain property 'a'
================================================== prop_access_on_non_object
exit_code: 103
--------------------------------------------------
[].a;
--------------------------------------------------
--------------------------------------------------
runtime_errors/prop_access_on_non_object.sd:1:1: properties can only be accessed on objects, got 'list'
================================================== prop_assign_on_non_object
exit_code: 103
--------------------------------------------------
[].a = 1;
--------------------------------------------------
--------------------------------------------------
runtime_errors/prop_assign_on_non_object.sd:1:1: properties can only be accessed on objects, got 'list'
================================================== negative_string_index
exit_code: 103
--------------------------------------------------
"abc"[-1];
--------------------------------------------------
--------------------------------------------------
runtime_errors/negative_string_index.sd:1:7: index can't be negative
================================================== negative_list_index
exit_code: 103
--------------------------------------------------
[1, 2, 3][-1];
--------------------------------------------------
--------------------------------------------------
runtime_errors/negative_list_index.sd:1:11: index can't be negative
================================================== value_not_range_indexable
exit_code: 103
--------------------------------------------------
1[2:];
--------------------------------------------------
--------------------------------------------------
runtime_errors/value_not_range_indexable.sd:1:1: only 'list's or 'string's can be range-indexed
================================================== value_not_range_index_assignable
exit_code: 103
--------------------------------------------------
1[2:] = [];
--------------------------------------------------
--------------------------------------------------
runtime_errors/value_not_range_index_assignable.sd:1:1: only 'list's can update range indices
================================================== range_index_assign_on_non_indexable
exit_code: 103
--------------------------------------------------
[][2:] = 1;
--------------------------------------------------
--------------------------------------------------
runtime_errors/range_index_assign_on_non_indexable.sd:1:1: only 'list's or 'string's can be assigned to range indexes, got 'int'
================================================== range_start_out_of_list_bounds
exit_code: 103
--------------------------------------------------
[][2:] = [];
--------------------------------------------------
--------------------------------------------------
runtime_errors/range_start_out_of_list_bounds.sd:1:1: range start (2) is greater than list length (0)
================================================== range_start_not_before_end
exit_code: 103
--------------------------------------------------
[1, 2, 3][2:1] = [1, 2, 3];
--------------------------------------------------
--------------------------------------------------
runtime_errors/range_start_not_before_end.sd:1:1: range end (1) must be greater than range start (2)
================================================== range_end_out_of_list_bounds
exit_code: 103
--------------------------------------------------
[][:2] = [];
--------------------------------------------------
--------------------------------------------------
runtime_errors/range_end_out_of_list_bounds.sd:1:1: range end (2) is greater than list length (0)
================================================== negative_string_range_start
exit_code: 103
--------------------------------------------------
"abc"[-1:];
--------------------------------------------------
--------------------------------------------------
runtime_errors/negative_string_range_start.sd:1:7: index can't be negative
================================================== negative_string_range_end
exit_code: 103
--------------------------------------------------
"abc"[:-1];
--------------------------------------------------
--------------------------------------------------
runtime_errors/negative_string_range_end.sd:1:8: index can't be negative
================================================== bad_string_range_start
exit_code: 103
--------------------------------------------------
"abc"[5:];
--------------------------------------------------
--------------------------------------------------
runtime_errors/bad_string_range_start.sd:1:1: range [5:3] is outside the string bounds
================================================== bad_string_range_end
exit_code: 103
--------------------------------------------------
"abc"[:5];
--------------------------------------------------
--------------------------------------------------
runtime_errors/bad_string_range_end.sd:1:1: range [0:5] is outside the string bounds
================================================== negative_list_range_start
exit_code: 103
--------------------------------------------------
[1][-1:];
--------------------------------------------------
--------------------------------------------------
runtime_errors/negative_list_range_start.sd:1:5: index can't be negative
================================================== negative_list_range_end
exit_code: 103
--------------------------------------------------
[1][:-1];
--------------------------------------------------
--------------------------------------------------
runtime_errors/negative_list_range_end.sd:1:6: index can't be negative
================================================== bad_list_range_start
exit_code: 103
--------------------------------------------------
[1][5:];
--------------------------------------------------
--------------------------------------------------
runtime_errors/bad_list_range_start.sd:1:1: range [5:1] is outside the list bounds
================================================== bad_list_range_end
exit_code: 103
--------------------------------------------------
[1][:5];
--------------------------------------------------
--------------------------------------------------
runtime_errors/bad_list_range_end.sd:1:1: range [0:5] is outside the list bounds
================================================== object_prop_shorthand_not_var
exit_code: 103
--------------------------------------------------
{1};
--------------------------------------------------
--------------------------------------------------
runtime_errors/object_prop_shorthand_not_var.sd:1:2: object property name isn't a variable
================================================== object_prop_shorthand_undefined
exit_code: 103
--------------------------------------------------
{value};
--------------------------------------------------
--------------------------------------------------
runtime_errors/object_prop_shorthand_undefined.sd:1:2: 'value' is not defined
================================================== spread_non_list_in_list
exit_code: 103
--------------------------------------------------
[1..];
--------------------------------------------------
--------------------------------------------------
runtime_errors/spread_non_list_in_list.sd:1:2: only lists can be spread in lists, got 'int'
================================================== spread_non_object_in_object
exit_code: 103
--------------------------------------------------
{1..};
--------------------------------------------------
--------------------------------------------------
runtime_errors/spread_non_object_in_object.sd:1:2: only objects can be spread in objects, got 'int'
================================================== string_index_assign
exit_code: 103
--------------------------------------------------
"abc"[1] = "d";
--------------------------------------------------
--------------------------------------------------
runtime_errors/string_index_assign.sd:1:1: only 'list's or 'object's can update indices
================================================== object_destruct_on_non_object
exit_code: 103
--------------------------------------------------
{a} = 1;
--------------------------------------------------
--------------------------------------------------
runtime_errors/object_destruct_on_non_object.sd:1:1: only objects can be destructured into objects, got 'int'
================================================== spread_on_object_destruct
exit_code: 103
--------------------------------------------------
{xs..} = {};
--------------------------------------------------
--------------------------------------------------
runtime_errors/spread_on_object_destruct.sd:1:2: can't use spread operator in object destructuring
================================================== object_assign_prop_shorthand_not_var
exit_code: 103
--------------------------------------------------
{} = {1};
--------------------------------------------------
--------------------------------------------------
runtime_errors/object_assign_prop_shorthand_not_var.sd:1:7: object property name isn't a variable
================================================== object_destruct_prop_not_found
exit_code: 103
--------------------------------------------------
{a, b, c} := {"a": 1, "b": 2};
--------------------------------------------------
--------------------------------------------------
runtime_errors/object_destruct_prop_not_found.sd:1:8: object doesn't contain property 'c'
================================================== object_destruct_dup_name1
exit_code: 103
--------------------------------------------------
{a, a} := {"a": 1, "b": 2};
--------------------------------------------------
--------------------------------------------------
runtime_errors/object_destruct_dup_name1.sd:1:5: 'a' is bound multiple times in this binding
================================================== object_destruct_dup_name2
exit_code: 103
--------------------------------------------------
{a, "b": a} := {"a": 1, "b": 2};
--------------------------------------------------
--------------------------------------------------
runtime_errors/object_destruct_dup_name2.sd:1:10: 'a' is bound multiple times in this binding
================================================== list_destruct_dup_name
exit_code: 103
--------------------------------------------------
[a, a] := [1, 2];
--------------------------------------------------
--------------------------------------------------
runtime_errors/list_destruct_dup_name.sd:1:5: 'a' is bound multiple times in this binding
================================================== cannot_mix_var_declarations2
exit_code: 103
--------------------------------------------------
a := 1;
{a, b} := {"a": 1, "b": 2};
--------------------------------------------------
--------------------------------------------------
runtime_errors/cannot_mix_var_declarations2.sd:2:2: 'a' is already defined in the current scope at [1:1]
================================================== object_destruct_already_in_scope
exit_code: 103
--------------------------------------------------
b := 3;
{"a": b} := {"a": 1, "b": 2};
--------------------------------------------------
--------------------------------------------------
runtime_errors/object_destruct_already_in_scope.sd:2:7: 'b' is already defined in the current scope at [1:1]
================================================== list_destruct_on_non_list
exit_code: 103
--------------------------------------------------
[a] = 1;
--------------------------------------------------
--------------------------------------------------
runtime_errors/list_destruct_on_non_list.sd:1:1: only lists can be destructured into lists, got 'int'
================================================== list_destruct_item_mismatch
exit_code: 103
--------------------------------------------------
[a, b] := [1, 2, 3];
--------------------------------------------------
--------------------------------------------------
runtime_errors/list_destruct_item_mismatch.sd:1:1: cannot bind 3 item(s) to 2 variable name(s)
================================================== list_collect_too_few
exit_code: 103
--------------------------------------------------
[a, b, ..c] := [1];
--------------------------------------------------
--------------------------------------------------
runtime_errors/list_collect_too_few.sd:1:1: cannot bind 1 item(s) to 3 variable name(s)
================================================== list_destruct_spread
exit_code: 103
--------------------------------------------------
[a..] := [1];
--------------------------------------------------
--------------------------------------------------
runtime_errors/list_destruct_spread.sd:1:1: cannot use spread operator (at index 0) of list destructure
================================================== list_collect_outside_destruct
exit_code: 103
--------------------------------------------------
[a] := [..1];
--------------------------------------------------
--------------------------------------------------
runtime_errors/list_collect_outside_destruct.sd:1:8: cannot collect 'list' items outside a destructure
================================================== param_destruct_spread
exit_code: 103
--------------------------------------------------
fn f(a, b, ..c) {
    print(c);
}
f(1);
--------------------------------------------------
--------------------------------------------------
runtime_errors/param_destruct_spread.sd:4:1: expected at least 2 arguments, got 1
==================================================
